sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/CharVarcharUtils.scala

```diff
@@ -52,6 +54,15 @@ object CharVarcharUtils {
     dt.existsRecursively(f => f.isInstanceOf[CharType] || f.isInstanceOf[VarcharType])
   }
 
+  /**
+   * Validate the given [[DataType]] to fail if it is char or varchar types or contains nested ones
```

Comment:
`... if the given data type is CharType/VarcharType or has nested CharType/VarcharType.`